lib/gitWorkflow.js

```diff
@@ -71,9 +55,9 @@ class GitWorkflow {
   async backupMergeStatus() {
     debug('Backing up merge state...')
     await Promise.all([
-      readBufferFromFile(this.mergeHeadFilename).then(buffer => (this.mergeHeadBuffer = buffer)),
-      readBufferFromFile(this.mergeModeFilename).then(buffer => (this.mergeModeBuffer = buffer)),
-      readBufferFromFile(this.mergeMsgFilename).then(buffer => (this.mergeMsgBuffer = buffer))
+      readFile(this.mergeHeadFilename).then(buffer => (this.mergeHeadBuffer = buffer)),
```

Comment:
Why not use `await` here?

Reply:
`Promise.all` is awaited, and the use of `then` resulted in much simpler notation IMO.

Reply:
I’m still not convinced this is better than `this.mergeHeader = await readFile()`

Reply:
Can you show an example, maybe I’m not getting something?

Reply:
Ah I missed `Promise.all` but. Yet I’m wondering if `await` automatically returns a promise. Not sure if that’s going to be a valid syntax if put there.

Reply:
If I understood correctly, your suggestion would look like this:

```js
async backupMergeStatus() {
    debug('Backing up merge state...')
    this.mergeHeadBuffer = await readFile(this.mergeHeadFilename)
    this.mergeModeBuffer = await readFile(this.mergeModeFilename)
    this.mergeMsgBuffer = await readFile(this.mergeMsgFilename)
    debug('Done backing up merge state!')
  }
```

and

```js
async restoreMergeStatus() {
    debug('Restoring merge state...')
    try {
      if (this.mergeHeadBuffer) await writeFile(this.mergeHeadFilename, this.mergeHeadBuffer)
      if (this.mergeModeBuffer) await writeFile(this.mergeModeFilename, this.mergeModeBuffer)
      if (this.mergeMsgBuffer) await writeFile(this.mergeMsgFilename, this.mergeMsgBuffer)
      debug('Done restoring merge state!')
    } catch (error) {
      debug('Failed restoring merge state with error:')
      debug(error)
      throw new Error('Merge state could not be restored due to an error!')
    }
  }
```

This looks cleaner but `await` will mean that the files are read/written one after other. Might not mean anything "in real life", but I originally chose `Promise.all` for concurrency support. We also use `Promise.all` over arrays mapped to promises in five other places in `gitWorkflow`.

Reply:
Yeah let’s keep it like it is